readthedocs/core/resolver.py

```diff
@@ -93,11 +92,12 @@ def resolve_path(self, project, filename='', version_slug=None,
 
         filename = self._fix_filename(project, filename)
 
-        if main_language_project:
-            project_slug = main_language_project.slug
+        if main_language_project_id:
+            project_slug = project.main_language_project.slug
             language = project.language
             subproject_slug = None
-        elif relation:
+        elif relations.count():
```

Comment:
Can this be `exists()`?

Reply:
There's a couple approaches:

- Either `project.superprojects.all` is cached due to a prefetch and then we can just use `len()`
- The superproject could be passed into this function (or the domain and then we don't need to get the superproject)

Running `project.superprojects.all().count()` will result in a database query which is exactly what we're trying to avoid. Perhaps it doesn't do that when the queryset is cached. I'm not 100% sure.